account_invoice_margin/models/account_invoice.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# © 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models
+import openerp.addons.decimal_precision as dp
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    margin = fields.Float(
+        compute='_compute_margin',
+        digits=dp.get_precision('Product Price'),
+        store=True,
+        string='Margin',
+    )
+    margin_percent = fields.Float(
+        string='% Margin',
+        compute='_compute_margin',
+        store=True,
+        readonly=True,
+    )
+    purchase_price = fields.Float(
+        digits=dp.get_precision('Product Price'),
+        string='Cost',
+    )
+
+    @api.multi
+    @api.depends('purchase_price', 'price_subtotal')
+    def _compute_margin(self):
+        for line in self.filtered(lambda x: x.invoice_id.type[:2] != 'in'):
+            tmp_margin = line.price_subtotal - (
+                line.purchase_price * line.quantity)
+            if line.invoice_id.type == 'out_refund':
+                tmp_margin *= -1
```

Comment:
Better to make `tmp_margin = -tmp_margin`

Reply:
The idea for making this is to get proper results on invoice analysis, so there's 2 options:

* Keep it as is.
* Have 2 fields: `margin` and `margin_signed`, both being computed here, but one used on invoice view and the other on the analysis.